src/screening/bloodSugar/BloodSugarTestForm.jsx

```diff
@@ -5,39 +5,36 @@ import { Grid, Row, FormGroup, ControlLabel, Col } from 'react-bootstrap';
 import { ENCOUNTER_TYPES, CONCEPTS, FORM_ANSWERS} from "../../constants";
 import '../../assets/css/tabs.css';
 import ScreeningForm from "../ScreeningForm";
-import { formValueSelector } from "redux-form";
+import {change, formValueSelector, untouch} from "redux-form";
 import { noPaddingLeftAndRight, flexBaseline } from "../../pwaStyles";
 
 
 class BloodSugarTestForm extends React.PureComponent {
 
+  componentDidUpdate(prevProps) {
+    if (typeof this.props.fsTestType.value !== 'undefined' &&
+      typeof prevProps.fsTestType.value !== 'undefined' &&
+      this.props.fsTestType.value !== prevProps.fsTestType.value) {
+      this.clearField(this.props.bloodGlucose.fieldName);
+    }
+  }
+
+  clearField(field) {
+    this.props.dispatch(change(this.props.formInstanceId, field, null));
+    this.props.dispatch(untouch(this.props.formInstanceId, field));
+  }
+
   render() {
 
     const {
       fsTestType
     } = this.props;
 
+    let glucoseConcept = Object.assign({}, CONCEPTS.BLOOD_GLUCOSE);
 
-    let fastingConcept = Object.assign({}, CONCEPTS.RANDOM_BLOOD_GLUCOSE, { hiNormal: 126 });
-
-
-    const randomObs = (
-      <Obs
-        concept={ CONCEPTS.RANDOM_BLOOD_GLUCOSE }
-        placeholder="value"
-        path="random-glucose"
-        required={ true }
-      />
-    );
-
-    const fastingObs = (
-      <Obs
-        concept={ fastingConcept }
-        placeholder="value"
-        path="fasting-glucose"
-        required={ true }
-      />
-    );
```

Comment:
So @cioan I went back to using a single obs tag, though I was never able to figure out why the way you had it wasn't working! One issue I did see with the two obs was that if you saved the form and when went back and changed it, you ended up with both obs behind the scene.